src/sentry/static/sentry/app/views/integrationOrganizationLink.tsx

```diff
@@ -0,0 +1,285 @@
+import {RouteComponentProps} from 'react-router/lib/Router';
+import React from 'react';
+import styled from '@emotion/styled';
+import {urlEncode} from '@sentry/utils';
+import {components} from 'react-select';
+
+import {Organization, IntegrationProvider} from 'app/types';
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {t, tct} from 'app/locale';
+import {
+  trackIntegrationEvent,
+  getIntegrationFeatureGate,
+} from 'app/utils/integrationUtil';
+import Alert from 'app/components/alert';
+import AsyncView from 'app/views/asyncView';
+import Button from 'app/components/button';
+import Field from 'app/views/settings/components/forms/field';
+import NarrowLayout from 'app/components/narrowLayout';
+import SelectControl from 'app/components/forms/selectControl';
+import IdBadge from 'app/components/idBadge';
+import LoadingIndicator from 'app/components/loadingIndicator';
+
+//installationId present for Github flow
+type Props = RouteComponentProps<{integrationSlug: string; installationId?: string}, {}>;
+
+type State = AsyncView['state'] & {
+  selectedOrgSlug?: string;
+  organization?: Organization;
+  provider?: IntegrationProvider;
+};
+
+export default class IntegrationOrganizationLink extends AsyncView<Props, State> {
+  getEndpoints(): [string, string][] {
+    return [['organizations', '/organizations/']];
+  }
+
+  getTitle() {
+    return t('Choose Installation Organization');
+  }
+
+  trackOpened() {
+    const {organization, provider} = this.state;
+    //should have these set but need to make TS happy
+    if (!organization || !provider) {
+      return;
+    }
+
+    trackIntegrationEvent(
+      {
+        eventKey: 'integrations.integration_viewed',
+        eventName: 'Integrations: Integration Viewed',
+        integration_type: 'first_party',
+        integration: provider.key,
+        //We actually don't know if it's installed but neither does the user in the view and multiple installs is possible
+        already_installed: false,
+        view: 'external_install',
+      },
+      organization,
```

Comment:
I wouldn't mind pulling out these parameters into a `get baseTrackPrams`

```tsx
get baseTrackParams() {
    const {organization, provider} = this.state;
    //should have these set but need to make TS happy
    if (!organization || !provider) {
      return;
    }

    return {
      eventKey: 'integrations.integration_viewed',
      eventName: 'Integrations: Integration Viewed',
      integration_type: 'first_party',
      integration: provider.key,
      // We actually don't know if it's installed but neither does
      // the user in the view and multiple installs is possible
      already_installed: false,
      view: 'external_install',
    } as const;
}

trackOpened() {
  const {organization, provider} = this.state;
  //should have these set but need to make TS happy
  if (!organization || !provider) {
    return;
  }

  trackIntegrationEvent(this.baseTrackParams, organization);
}
```

Reply:
Since you use them twice that is

Reply:
@EvanPurkhiser good suggestion, I'll make this change